Tests/Integration.Tests.ps1

```diff
@@ -0,0 +1,97 @@
+            Invoke-Command $startRegistryCommand
+        }
+
+        AfterAll {
+            Invoke-Command 'docker container stop registry'
+            Invoke-Command 'docker container rm -v registry'
+        }
+
+        BeforeEach {
+            $script:backupDebugPref = $DebugPreference
+            $DebugPreference='Continue'
+            $script:backupLocation = Get-Location
+        }
+
+        AfterEach {
+            $DebugPreference=$script:DebugPreference
+            Set-Location $script:backupLocation
+        }
+
+        It "Use case #1: can derive docker image name and tag in one go" {
+            $exampleReposPath = Join-Path $testData "ExampleRepos"
+            $location = Join-Path $exampleReposPath "3.0/servercore/amd64"
+            Set-Location $location
+            New-FakeGitRepository $location
+
+            $result = Find-ImageName $location
+
+            $result = Format-DockerTag | Invoke-DockerBuild -ImageName $result.ImageName
+
+            $result.Dockerfile | Should -BeLike "*Dockerfile"
+            $result.ImageName | Should -Be "dockerbuild-pwsh"
+            $result.Tag | Should -Be  "3.0-servercore-amd64"
+        }
+
+        It 'supports docker login' {
+            $result = Invoke-DockerLogin -Username 'admin' -Password (ConvertTo-SecureString 'password' –asplaintext –force) -Registry 'localhost:5000'
+            $result.ExitCode | Should -Be 0
+        }
+
+        It "Use case #2: can build and push in one go" {
+
+            $exampleReposPath = Join-Path $testData "ExampleRepos"
+            $location = Join-Path $exampleReposPath "3.0/servercore/amd64"
+            Set-Location $location
+            New-FakeGitRepository $location
+
+            Invoke-DockerLogin -Username 'admin' -Password (ConvertTo-SecureString 'password' –asplaintext –force) -Registry 'localhost:5000'
+            Invoke-DockerBuild -ImageName 'integration-testcase-2' -Registry 'localhost:5000' | Invoke-DockerPush -Registry 'localhost:5000'
+
+            $result = Invoke-DockerPull -Registry 'localhost:5000' -ImageName 'integration-testcase-2' -Tag 'latest'
+            Write-Debug $result.Output
+            $result.ExitCode | Should -Be 0
+        }
+
+        # It 'can login' {
+        #     $result = Invoke-DockerLogin    -Registry 'localhost:5000' `
+        #                                     -Username 'admin' `
+        #                                     -Password (ConvertTo-SecureString 'password' –asplaintext –force)
+
+        #     $result | Should -Not -BeNullOrEmpty
+        #     $result.ExitCode | Should -Be 0
+        # }
+
+        # It "can build and push my image with 'latest' tag" {
+        #     $dockerFile = Join-Path $dockerImages 'Dockerfile'
+        #     # Invoke-DockerBuild -Image 'integrationtest'
+        #     # 1. Login
+        #     # 2. Pull the image
+        #     # 3. Lint
+        #     # 4. Build
+        #     # 5. Test
+        #     # 6. Tag
+        #     # 7. Push
```

Comment:
Hmmm this seems to be mostly implemented.

Reply:
Yup.